### PR TITLE
Add table aws_ec2_managed_prefix_list. Closes #803

### LINKED ISSUE #803: Add table aws_ec2_managed_prefix_list
**References**
https://github.com/turbot/steampipe-plugin-aws/issues/799 and https://github.com/turbot/steampipe-plugin-aws/issues/800

### PR DESCRIPTION
**Pull request by c0d3r-arnab**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
arnab@turbotindias-MacBook-Pro aws-test % ./tint.js aws_ec2_managed_prefix_list
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_ec2_managed_prefix_list []

PRETEST: tests/aws_ec2_managed_prefix_list

TEST: tests/aws_ec2_managed_prefix_list
Running terraform

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aws_ec2_managed_prefix_list.named_test_resource will be created
  + resource "aws_ec2_managed_prefix_list" "named_test_resource" {
      + address_family = "IPv4"
      + arn            = (known after apply)
      + id             = (known after apply)
      + max_entries    = 5
      + name           = "turbottest34102"
      + owner_id       = (known after apply)
      + tags           = {
          + "name" = "turbottest34102"
        }
      + tags_all       = {
          + "name" = "turbottest34102"
        }
      + version        = (known after apply)

      + entry {
          + cidr        = (known after apply)
          + description = (known after apply)
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + address_family = "IPv4"
  + arn            = (known after apply)
  + id             = (known after apply)
  + owner_id       = (known after apply)
  + resource_name  = "turbottest34102"
aws_ec2_managed_prefix_list.named_test_resource: Creating...
aws_ec2_managed_prefix_list.named_test_resource: Creation complete after 6s [id=pl-06b9d15a971a9177e]

Warning: Deprecated Resource

  with data.null_data_source.resource,
  on variables.tf line 43, in data "null_data_source" "resource":
  43: data "null_data_source" "resource" {

The null_data_source was historically used to construct intermediate values
to re-use elsewhere in configuration, the same can now be achieved using
locals

(and one more similar warning elsewhere)

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

address_family = "IPv4"
arn = "arn:aws:ec2:us-east-1:632901234528:prefix-list/pl-06b9d15a971a9177e"
id = "pl-06b9d15a971a9177e"
owner_id = "632901234528"
resource_name = "turbottest34102"

Running SQL query: test-list-query.sql
[
  {
    "address_family": "IPv4",
    "arn": "arn:aws:ec2:us-east-1:632901234528:prefix-list/pl-06b9d15a971a9177e",
    "id": "pl-06b9d15a971a9177e",
    "name": "turbottest34102",
    "owner_id": "632901234528"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:ec2:us-east-1:632901234528:prefix-list/pl-06b9d15a971a9177e"
    ],
    "tags": {
      "name": "turbottest34102"
    },
    "title": "turbottest34102"
  }
]
✔ PASSED

POSTTEST: tests/aws_ec2_managed_prefix_list

TEARDOWN: tests/aws_ec2_managed_prefix_list

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  name,
  id,
  arn,
  state,
  owner_id
from
  aws_ec2_managed_prefix_list;
+---------------------------------------+----------------------+---------------------------------------------------------------------+-----------------+--------------+
| name                                  | id                   | arn                                                                 | state           | owner_id     |
+---------------------------------------+----------------------+---------------------------------------------------------------------+-----------------+--------------+
| com.amazonaws.ap-south-1.dynamodb     | pl-66a7420f          | arn:aws:ec2:ap-south-1:aws:prefix-list/pl-66a7420f                  | create-complete | AWS          |
| com.amazonaws.ap-south-1.s3           | pl-78a54011          | arn:aws:ec2:ap-south-1:aws:prefix-list/pl-78a54011                  | create-complete | AWS          |
| com.amazonaws.us-east-1.s3            | pl-63a5400a          | arn:aws:ec2:us-east-1:aws:prefix-list/pl-63a5400a                   | create-complete | AWS          |
| testPrefix                            | pl-03a3e735e3467c0c4 | arn:aws:ec2:us-east-1:632901234528:prefix-list/pl-03a3e735e3467c0c4 | create-complete | 632901234528 |
| com.amazonaws.us-east-1.dynamodb      | pl-02cd2c6b          | arn:aws:ec2:us-east-1:aws:prefix-list/pl-02cd2c6b                   | create-complete | AWS          |
| com.amazonaws.ap-southeast-1.s3       | pl-6fa54006          | arn:aws:ec2:ap-southeast-1:aws:prefix-list/pl-6fa54006              | create-complete | AWS          |
| com.amazonaws.ap-southeast-1.dynamodb | pl-67a5400e          | arn:aws:ec2:ap-southeast-1:aws:prefix-list/pl-67a5400e              | create-complete | AWS          |
| com.amazonaws.eu-west-3.dynamodb      | pl-abb451c2          | arn:aws:ec2:eu-west-3:aws:prefix-list/pl-abb451c2                   | create-complete | AWS          |
| com.amazonaws.eu-west-3.s3            | pl-23ad484a          | arn:aws:ec2:eu-west-3:aws:prefix-list/pl-23ad484a                   | create-complete | AWS          |
| com.amazonaws.eu-west-2.s3            | pl-7ca54015          | arn:aws:ec2:eu-west-2:aws:prefix-list/pl-7ca54015                   | create-complete | AWS          |
| com.amazonaws.eu-west-2.dynamodb      | pl-b3a742da          | arn:aws:ec2:eu-west-2:aws:prefix-list/pl-b3a742da                   | create-complete | AWS          |
| com.amazonaws.eu-central-1.s3         | pl-6ea54007          | arn:aws:ec2:eu-central-1:aws:prefix-list/pl-6ea54007                | create-complete | AWS          |
| com.amazonaws.eu-central-1.dynamodb   | pl-66a5400f          | arn:aws:ec2:eu-central-1:aws:prefix-list/pl-66a5400f                | create-complete | AWS          |
| com.amazonaws.ap-northeast-1.dynamodb | pl-78a54011          | arn:aws:ec2:ap-northeast-1:aws:prefix-list/pl-78a54011              | create-complete | AWS          |
| com.amazonaws.ap-northeast-1.s3       | pl-61a54008          | arn:aws:ec2:ap-northeast-1:aws:prefix-list/pl-61a54008              | create-complete | AWS          |
| com.amazonaws.ap-northeast-2.s3       | pl-78a54011          | arn:aws:ec2:ap-northeast-2:aws:prefix-list/pl-78a54011              | create-complete | AWS          |
| com.amazonaws.eu-north-1.s3           | pl-c3aa4faa          | arn:aws:ec2:eu-north-1:aws:prefix-list/pl-c3aa4faa                  | create-complete | AWS          |
| com.amazonaws.eu-north-1.dynamodb     | pl-adae4bc4          | arn:aws:ec2:eu-north-1:aws:prefix-list/pl-adae4bc4                  | create-complete | AWS          |
| com.amazonaws.ap-northeast-2.dynamodb | pl-48a54021          | arn:aws:ec2:ap-northeast-2:aws:prefix-list/pl-48a54021              | create-complete | AWS          |
| com.amazonaws.ca-central-1.dynamodb   | pl-4ea54027          | arn:aws:ec2:ca-central-1:aws:prefix-list/pl-4ea54027                | create-complete | AWS          |
| com.amazonaws.us-west-1.s3            | pl-6ba54002          | arn:aws:ec2:us-west-1:aws:prefix-list/pl-6ba54002                   | create-complete | AWS          |
| com.amazonaws.ca-central-1.s3         | pl-7da54014          | arn:aws:ec2:ca-central-1:aws:prefix-list/pl-7da54014                | create-complete | AWS          |
| com.amazonaws.us-west-1.dynamodb      | pl-6ea54007          | arn:aws:ec2:us-west-1:aws:prefix-list/pl-6ea54007                   | create-complete | AWS          |
| com.amazonaws.us-west-2.dynamodb      | pl-00a54069          | arn:aws:ec2:us-west-2:aws:prefix-list/pl-00a54069                   | create-complete | AWS          |
| com.amazonaws.us-west-2.s3            | pl-68a54001          | arn:aws:ec2:us-west-2:aws:prefix-list/pl-68a54001                   | create-complete | AWS          |
| com.amazonaws.eu-west-1.dynamodb      | pl-6fa54006          | arn:aws:ec2:eu-west-1:aws:prefix-list/pl-6fa54006                   | create-complete | AWS          |
| com.amazonaws.us-east-2.s3            | pl-7ba54012          | arn:aws:ec2:us-east-2:aws:prefix-list/pl-7ba54012                   | create-complete | AWS          |
| com.amazonaws.eu-west-1.s3            | pl-6da54004          | arn:aws:ec2:eu-west-1:aws:prefix-list/pl-6da54004                   | create-complete | AWS          |
| com.amazonaws.ap-southeast-2.s3       | pl-6ca54005          | arn:aws:ec2:ap-southeast-2:aws:prefix-list/pl-6ca54005              | create-complete | AWS          |
| com.amazonaws.ap-southeast-2.dynamodb | pl-62a5400b          | arn:aws:ec2:ap-southeast-2:aws:prefix-list/pl-62a5400b              | create-complete | AWS          |
| com.amazonaws.us-east-2.dynamodb      | pl-4ca54025          | arn:aws:ec2:us-east-2:aws:prefix-list/pl-4ca54025                   | create-complete | AWS          |
| com.amazonaws.sa-east-1.s3            | pl-6aa54003          | arn:aws:ec2:sa-east-1:aws:prefix-list/pl-6aa54003                   | create-complete | AWS          |
| com.amazonaws.sa-east-1.dynamodb      | pl-60a54009          | arn:aws:ec2:sa-east-1:aws:prefix-list/pl-60a54009                   | create-complete | AWS          |
+---------------------------------------+----------------------+---------------------------------------------------------------------+-----------------+--------------+
> select
  name,
  id,
  arn,
  state,
  owner_id
from
  aws_ec2_managed_prefix_list
where
  owner_id <> 'AWS';
+------------+----------------------+---------------------------------------------------------------------+-----------------+--------------+
| name       | id                   | arn                                                                 | state           | owner_id     |
+------------+----------------------+---------------------------------------------------------------------+-----------------+--------------+
| testPrefix | pl-03a3e735e3467c0c4 | arn:aws:ec2:us-east-1:632901234528:prefix-list/pl-03a3e735e3467c0c4 | create-complete | 632901234528 |
+------------+----------------------+---------------------------------------------------------------------+-----------------+--------------+
> select
  name,
  id,
  address_family
from
  aws_ec2_managed_prefix_list
where
  address_family = 'IPv6';
+------------+----------------------+----------------+
| name       | id                   | address_family |
+------------+----------------------+----------------+
| testPrefix | pl-03a3e735e3467c0c4 | IPv6           |
+------------+----------------------+----------------+
> select
  name,
  id,
  arn,
  state,
  owner_id
from
  aws_ec2_managed_prefix_list
where
  id in ('pl-03a3e735e3467c0c4', 'pl-4ca54025');
+----------------------------------+----------------------+---------------------------------------------------------------------+-----------------+--------------+
| name                             | id                   | arn                                                                 | state           | owner_id     |
+----------------------------------+----------------------+---------------------------------------------------------------------+-----------------+--------------+
| testPrefix                       | pl-03a3e735e3467c0c4 | arn:aws:ec2:us-east-1:632901234528:prefix-list/pl-03a3e735e3467c0c4 | create-complete | 632901234528 |
| com.amazonaws.us-east-2.dynamodb | pl-4ca54025          | arn:aws:ec2:us-east-2:aws:prefix-list/pl-4ca54025                   | create-complete | AWS          |
+----------------------------------+----------------------+---------------------------------------------------------------------+-----------------+--------------+
> select
  name,
  id,
  arn,
  state,
  owner_id
from
  aws_ec2_managed_prefix_list
where
  name in ('testPrefix', 'com.amazonaws.us-east-2.dynamodb');
+----------------------------------+----------------------+---------------------------------------------------------------------+-----------------+--------------+
| name                             | id                   | arn                                                                 | state           | owner_id     |
+----------------------------------+----------------------+---------------------------------------------------------------------+-----------------+--------------+
| testPrefix                       | pl-03a3e735e3467c0c4 | arn:aws:ec2:us-east-1:632901234528:prefix-list/pl-03a3e735e3467c0c4 | create-complete | 632901234528 |
| com.amazonaws.us-east-2.dynamodb | pl-4ca54025          | arn:aws:ec2:us-east-2:aws:prefix-list/pl-4ca54025                   | create-complete | AWS          |
+----------------------------------+----------------------+---------------------------------------------------------------------+-----------------+--------------+
> select
  name,
  id,
  arn,
  state,
  owner_id
from
  aws_ec2_managed_prefix_list
where
  owner_id = '632901234528';
+------------+----------------------+---------------------------------------------------------------------+-----------------+--------------+
| name       | id                   | arn                                                                 | state           | owner_id     |
+------------+----------------------+---------------------------------------------------------------------+-----------------+--------------+
| testPrefix | pl-03a3e735e3467c0c4 | arn:aws:ec2:us-east-1:632901234528:prefix-list/pl-03a3e735e3467c0c4 | create-complete | 632901234528 |
+------------+----------------------+---------------------------------------------------------------------+-----------------+--------------+
```
</details>

### CODE REVIEW

File: aws/table_aws_ec2_managed_prefix_list.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "state_message",
+				Description: "The state message.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "version",
+				Description: "The state message.",
```
Comment (bigdatasourav) [inline aws/table_aws_ec2_managed_prefix_list.go:71]: Description is not correct.

File: docs/tables/aws_ec2_managed_prefix_list.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+select
+  name,
+  id,
+  arn,
+  state,
+  owner_id
+from
+  aws_ec2_managed_prefix_list
+where
+  owner_id = '632901234528';
```
Comment (misraved) [inline docs/tables/aws_ec2_managed_prefix_list.md:95]: Can we update this `owner_id` to use an arbitrary value ?

Comment (c0d3r-arnab) [reply docs/tables/aws_ec2_managed_prefix_list.md:95]: The ID isn't a valid one. I changed 4 digits of the owner_id

**Review by bigdatasourav — APPROVED** [review]
LGTM

File: aws/table_aws_ec2_managed_prefix_list.go [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+				Type:        proto.ColumnType_INT,
+			},
+			{
+				Name:        "owner_id",
+				Description: "The ID of the owner of the prefix list.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "state_message",
+				Description: "The state message.",
```
Comment (misraved) [inline aws/table_aws_ec2_managed_prefix_list.go:66]: Can we add a better description ?

**Review by misraved — COMMENTED** [review]
Make changes wherever applicable

**Review by misraved — APPROVED** [review]

**Pull request merged**